frontend/src/components/ConfigForm.vue

```diff
@@ -43,67 +79,120 @@
-        <p v-if="userAppCfg.deploymentMethod === 'deploymentManual'" class="underRadioField">
-          <a
-            href="https://github.com/bcgov/nr-get-token/wiki/WebADE-Access"
-            target="_blank"
-          >Instructions for manual deployment</a>
-        </p>
-        <v-radio label="Direct Deploy" value="deploymentDirect"></v-radio>
-        <v-text-field
-          v-if="userAppCfg.deploymentMethod === 'deploymentDirect'"
-          label="Password"
-          required
-          :value="userAppCfg.userEnteredPassword"
-          v-on:keyup.stop="updateAppCfgField('userEnteredPassword', $event.target.value)"
-          class="underRadioField"
-        ></v-text-field>
-      </v-radio-group>
+      <v-form v-model="step2Valid">
+        <v-radio-group
+          :value="userAppCfg.deploymentMethod"
+          v-on:change="updateAppCfgField('deploymentMethod', $event)"
+          :mandatory="true"
+        >
+          <v-radio
+            label="Manual commit to Bitbucket (deploy with Jenkins)"
+            value="deploymentManual"
+          ></v-radio>
+          <p v-if="userAppCfg.deploymentMethod === 'deploymentManual'" class="underRadioField">
+            <a
+              href="https://github.com/bcgov/nr-get-token/wiki/WebADE-Access"
+              target="_blank"
+            >Instructions for manual deployment</a>
+          </p>
+          <v-radio label="Direct Deploy" value="deploymentDirect"></v-radio>
+
+          <v-layout row wrap>
+            <v-flex xs12>
+              <v-text-field
+                v-if="userAppCfg.deploymentMethod === 'deploymentDirect'"
+                label="Password"
+                required
+                :value="userAppCfg.userEnteredPassword"
+                v-on:keyup.stop="updateAppCfgField('userEnteredPassword', $event.target.value)"
+                class="underRadioField"
+                :rules="passwordRules"
+                :append-icon="showPw ? 'visibility' : 'visibility_off'"
+                :type="showPw ? 'text' : 'password'"
+                hint="At least 8 characters"
+                :counter="60"
+                @click:append="showPw = !showPw"
+              ></v-text-field>
+            </v-flex>
+          </v-layout>
+        </v-radio-group>
 
-      <v-btn
-        v-if="userAppCfg.deploymentMethod === 'deploymentDirect'"
-        color="success"
-        @click="submitConfig"
-      >Submit</v-btn>
-      <v-btn flat @click="appConfigStep = 1">Back</v-btn>
+        <v-btn
+          v-if="userAppCfg.deploymentMethod === 'deploymentDirect'"
+          color="success"
+          @click="submitConfig"
+          :disabled="!step2Valid"
+        >Submit</v-btn>
+        <v-btn flat @click="appConfigStep = 1">Back</v-btn>
+      </v-form>
     </v-stepper-content>
   </v-stepper>
 </template>
 
 <script>
 import { mapGetters } from "vuex";
+import {isValidJson} from "../utils/utils.js"
```

Comment:
spaces around isValidJson
Also if there's a way to use the @ relative path instead of directory traversal via .. it would make it cleaner

Reply:
Ah yes, I'd done '@/utils/utils.js' in the unit test file but forgot to adjust here.